pyproject.toml

```diff
@@ -1,3 +1,58 @@
+[build-system]
+requires = ["setuptools>=61.0"]
+build-backend = "setuptools.build_meta"
+
+[project]
+name = "levanter"
+version = "0.0.1"
+authors = [
+  { name="David Hall", email="dlwh@cs.stanford.edu" },
+    {name="Ivan Zhou", email="ivanz@stanford.edu"}
+]
+description = "Scalable Training for Foundation Models with Named Tensors and Jax"
+readme = "README.md"
+requires-python = ">=3.10"
+classifiers = [
+    "Programming Language :: Python :: 3",
+    "License :: OSI Approved :: Apache Software License 2.0",
+    "Operating System :: Linux",
+    "Operating System :: MacOS :: MacOS X",
+    "Development Status :: 3 - Alpha",
+    "Intended Audience :: Science/Research",
+]
+
+dependencies = [
+    # we'll require that you install cuda yourself, since the extras vary by system.
+    # jax = {version = ">=0.4.10,<0.5.0"}
+    "equinox~=0.9.0",
+    "transformers>=4.22.0",
+    "optax",
+    "wandb",
+    "pyrallis>=0.3.1",
+    "pyarrow>=11.0.0",
+    "zstandard>=0.20.0",
+    "datasets==2.11.0",
+    "gcsfs==2023.3.0",
+    "braceexpand>=0.1.7",
+    "jmp>=0.0.3",
+    "fsspec==2023.3.0", # pin this to make gcsfs happy
+    "tensorstore>=0.1.36",
+    "pytimeparse>=1.1.8",
+    "humanfriendly==10.0",
+    "requests",
+    "safetensors[numpy]",
+    "matplotlib>=3.7.0",
+    "tblib>=1.7.0,<2.0.0",
+    "dataclasses_json",
+    "ray[default]",
+    "rich>=13",
+]
+
+[project.urls]
+"Homepage" = "https://github.com/stanford-crfm/levanter"
+"Bug Tracker" = "https://github.com/stanford-crfm/levanter/issues"
+
+
 [tool.black]
 line-length = 119
 target-version = ["py36", "py37", "py38"]
```

Comment:
According to the conversation [here](https://github.com/psf/black/issues/751), 3.6 through 3.8 are currently the same as far as Black is concerned, so either variant will do the same.
According to [this PR](https://github.com/mopidy/mopidy/pull/2042/files), Python 10 should be `py310`.

So maybe we can do: `["py38", "py39", "py310"]`?

Reply:
i think we only support python 310+ so i'll just require that